### PR TITLE
Feat/37 points distribution

### PR DESCRIPTION
**Pull request by baonguyen1904**

### CODE REVIEW

File: src/bg_tasks/restaking_point_calculation.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    user_percentage: Dict[str, float],
+    partner_name: str,
+):
+    """
+    Distribute points to users based on their share percentages.
+    """
+    for user in user_positions:
+        user_points = session.exec(
+            select(UserPoints)
+            .where(UserPoints.wallet_address == user.user_address)
```
Comment (baonguyen1904) [inline src/bg_tasks/restaking_point_calculation.py:92]: add vault_id to UserPoints

File: src/bg_tasks/restaking_point_calculation.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        session.add(point_distribution)
+
+    session.commit()
+
+
+def main():
+    # get all vaults that have VaultCategory = points
+    partners = [constants.RENZO, constants.ZIRCUIT]
+
+    for partner in partners:
```
Comment (baonguyen1904) [inline src/bg_tasks/restaking_point_calculation.py:183]: change vault instead of partner

File: src/core/config.py [hunk 3]
```diff
@@ -101,6 +101,11 @@ def assemble_cors_origins(cls, v: Union[str, List[str]]) -> Union[List[str], str
     NEXT_PUBLIC_ROCK_ONYX_USDT_VAULT_ADDRESS: Optional[str] = None
     NEXT_PUBLIC_USDC_ADDRESS: Optional[str] = None
 
+    # RESTAKING
+    RENZO_BASE_API_URL: Optional[str] = "https://app.renzoprotocol.com/api/"
+    ZIRCUIT_BASE_API_URL: Optional[str] = "https://stake.zircuit.com/api/"
+    KELPDAO_BASE_API_URL: Optional[str] = "https://app.renzoprotocol.com/"
```
Comment (baonguyen1904) [inline src/core/config.py:107]: change name

**Review by SandroBui — APPROVED** [review]

File: src/bg_tasks/restaking_point_calculation.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    # get all users who have points in the vault
+    user_positions = session.exec(
+        select(UserPortfolio)
+        .where(UserPortfolio.vault_id == vault.id)
+        .where(UserPortfolio.status == PositionStatus.ACTIVE)
+    ).all()
+
+    # calculate the percentage of shares for each user
+    total_deposit_amount = sum([user.init_deposit for user in user_positions])
+    user_percentage = {}
```
Comment (baonguyen1904) [inline src/bg_tasks/restaking_point_calculation.py]: move this logic to `distribute_points_to_users` to reduce the loop time

File: src/bg_tasks/restaking_point_calculation.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    """
+    for user in user_positions:
+        user_points = session.exec(
+            select(UserPoints)
+            .where(UserPoints.wallet_address == user.user_address)
+            .where(UserPoints.partner_name == partner_name)
+            .where(UserPoints.vault_id == vault_id)
+        ).first()
+        if user_points:
+            user_points.points += earned_points * user_percentage[user.user_address]
```
Comment (baonguyen1904) [inline src/bg_tasks/restaking_point_calculation.py]: insert audit record

**Pull request merged**